android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -647,6 +650,56 @@ class GeckoEngineSession(
         )
     }
 
+    /**
+     * See [EngineSession.requestProductAnalysis]
+     */
+    override fun requestProductAnalysis(
+        url: String,
+        onResult: (ShoppingProductAnalysis) -> Unit,
+        onException: (Throwable) -> Unit,
+    ) {
+        geckoSession.requestAnalysis(url).then({
+                response ->
+            if (response == null) {
+                logger.error(
+                    "Invalid value: unable to get analysis result from Gecko Engine.",
+                )
+                onException(
+                    java.lang.IllegalStateException(
+                        "Invalid value: unable to get analysis result from Gecko Engine.",
+                    ),
+                )
+                return@then GeckoResult()
+            }
+            val highlights = Highlight(
+                response.highlights?.quality,
+                response.highlights?.price,
+                response.highlights?.shipping,
+                response.highlights?.appearance,
+                response.highlights?.competitiveness,
+            )
+            val analysisResult = GeckoShoppingProductAnalysis(
+                response.productId ?: "",
```

Comment:
If `productId` is null, we shouldn't return an empty string here. I think this case should correspond to returning a different object - this seems like a case where needsAnalysis would be true. Check [this doc section](https://docs.google.com/document/d/1eppEnQKrSKavM3qmlm36WG7y8o2ggm-oxugnoSr2tpw/edit#heading=h.ff818tiik47) for reference.

Reply:
I'll make this field nullable then.